components/brave_ads/core/internal/common/strings/string_conversions_util.cc

```diff
@@ -18,4 +21,21 @@ std::string BoolToString(const bool value) {
   return value ? kTrue : kFalse;
 }
 
+std::vector<float> ConvertDelimitedStringToVector(
+    const std::string& string,
+    const std::string& delimiter) {
+  const std::vector<std::string> components = base::SplitString(
+      string, delimiter, base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+
+  std::vector<float> vector;
+  vector.reserve(components.size());
+  for (const auto& component : components) {
+    double value;
+    base::StringToDouble(component, &value);
+    vector.push_back(static_cast<float>(value));
```

Comment:
@ptjames @LorenzoMinto can we not store as doubles so that we do not have to cast to floats throughout the code base

Reply:
briefly discussed this @ptjames. The reason behind using floats instead of doubles it's mostly a matter of efficiency and memory-impact. We ship down the vectors as floats rather than doubles because the vocabulary can be quite large and each word embedding is 300d, so the total difference between total double size and total float size is considerable. We don't need double precision for any of the computations.

Reply:
Sounds good but can we check where we cast, i.e. GetNorm and see if we can change those functions to reduce the amount of casting

Reply:
yeah, I tried to minimise casts on vector data side. Now VectorData works only on floats (before it was mixed float/double)